geiger/src/graph/Edge.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cell from './Cell';
+import {EdgeTypes} from './EdgeTypes';
+import {computeNodeId, computeEdgeId, computeOutPortId,
+  computeInPortId, computeAudioParamPortId} from './label';
+
+/**
+ * @typedef {Object} NodesConnection
+ * @property {string} sourceNodeId
+ * @property {string} destinationNodeId
+ * @property {string} fromChannel
+ * @property {string} toChannel
+ */
+
+/**
+ * @typedef {Object} NodeParamConnection
+ * @property {string} sourceNodeId
+ * @property {string} destinationNodeId
+ * @property {string} fromChannel
+ * @property {string} destinationParamName
+ */
+
+/**
+ * @typedef {NodesConnection | NodeParamConnection} ConnectionMessage
```

Comment:
You might want to revisit this based on the event/data structure in the chromium CL.